### PR TITLE
Added WifiWebAuthApplet launching

### PR DESCRIPTION
**Pull request by p-sam**
This adds support for launch the WifiWebAuthApplet with an arbitrary start URL.

Credits to @misson20000 for ilia and the ipc dump that allowed me to get the struct and build the initial implementation

### CODE REVIEW

**Comment by yellows8** [comment]
So zero code-RE was done for this?

**Comment by p-sam** [comment]
Most informations came by looking at the ipc exchanges. I'm trying to find in the code which of the two urls is the one the applet polls for the "X-Organization: Nintendo" header, I will rename them both accordingly once figured.

**Comment by yellows8** [comment]
"applet polls ..." Uh that's unrelated AFAIK (?)

**Comment by p-sam** [comment]
Uh, I meant that it could periodically request one of the two urls and that it may close itself if the header is present. I'd need to add a comment regarding that, and name more precisely both urls in the struct.

**Comment by yellows8** [comment]
Still doubt it's related to any applet args...

File: nx/include/switch/applets/web_wifi.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @brief Wrapper for using the WifiWebAuthApplet LibraryApplet.
+ * @author p-sam
+ * @copyright libnx Authors
+ */
+#pragma once
+#include "../types.h"
+#include "../services/applet.h"
+
+typedef struct {
+    u8 unk1[0x4];
```
Comment (yellows8) [inline nx/include/switch/applets/web_wifi.h]: unk_x{offset}, same below.

**Comment by yellows8** [comment]
Was going to work on web-applet some point after finishing swkbd-inline - will probably update this post-merge later.

File: nx/include/switch/applets/web_wifi.h (hunk 1 above)
Comment (p-sam) [reply nx/include/switch/applets/web_wifi.h]: done

File: nx/source/applets/web_wifi.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    LibAppletArgs commonargs;
+    libappletArgsCreate(&commonargs, 0);
+    rc = libappletArgsPush(&commonargs, &holder);
+
+    if (R_SUCCEEDED(rc)) rc = libappletPushInData(&holder, &config->arg, sizeof(config->arg));
+
+    if (R_SUCCEEDED(rc)) rc = appletHolderStart(&holder);
+    
+    if (R_SUCCEEDED(rc)) {
```
[the remark this answers is not in the document]
Comment (p-sam) [reply nx/source/applets/web_wifi.c]: done

File: nx/include/switch/applets/web.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/**
+ * @file web_wifi.h
```
Comment (yellows8) [inline nx/include/switch/applets/web.h]: rename

Comment (p-sam) [reply nx/include/switch/applets/web.h]: done

File: nx/source/applets/web.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    memset(config, 0, sizeof(WebWifiConfig));
+    _webWifiUrlCreate(&config->arg.url1, url);
+    _webWifiUrlCreate(&config->arg.url2, url);
+}
+
+
+Result webWifiShow(WebWifiConfig* config) {
+    Result rc = 0;
+    AppletHolder holder;
+    AppletStorage storage;
```
Comment (yellows8) [inline nx/source/applets/web.c]: Remove since unused.

File: nx/source/applets/web.c [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    strncpy(argUrl->url, url, sizeof(argUrl->url)-1);
+}
+
+void webWifiCreate(WebWifiConfig* config, const char* url) {
+    memset(config, 0, sizeof(WebWifiConfig));
+    _webWifiUrlCreate(&config->arg.url1, url);
+    _webWifiUrlCreate(&config->arg.url2, url);
+}
+
+
```
Comment (yellows8) [inline nx/source/applets/web.c]: remove newline

File: nx/source/applets/web.c [hunk 6]
```diff
@@ -0,0 +1,47 @@
+#include <string.h>
+#include <malloc.h>
+#include "types.h"
+#include "result.h"
+#include "services/applet.h"
+#include "applets/libapplet.h"
+#include "applets/web.h"
+
+static void _webWifiUrlCreate(WebWifiPageArgUrl* argUrl, const char* url) {
+    strncpy(argUrl->url, url, sizeof(argUrl->url)-1);
```
Comment (fincs) [inline nx/source/applets/web.c]: Should memset argUrl to zero first

**Comment by fincs** [comment]
Hmm, if the official name is `WifiWebAuthApplet`, shouldn't the prefix be `WifiWebAuth` instead of `WebWifi`? Are there any other web browser applets? In that case, filename should also be changed from `web.h/c` to `wifi_web.h/c`.

**Comment by yellows8** [comment]
sdk-nso handles multiple web-applets under "nn::web::".

**Comment by fincs** [comment]
Ah I see.

**Comment by yellows8** [comment]
"So ..." yeah

**Pull request merged**